sql-cli/examples/_dags/advanced.py

```diff
@@ -0,0 +1,69 @@
+# NOTE: This is an auto-generated file. Please do not edit this file manually.
+from airflow import DAG
+from airflow.utils import timezone
+from airflow.utils.task_group import TaskGroup
+from astro import sql as aql
+from astro.sql.table import Metadata, Table
+
+with DAG(
+    dag_id="advanced",
+    start_date=timezone.parse("2022-09-20"),
+) as dag:
+    sub_dir_1 = TaskGroup(group_id="sub_dir_1")
+    sub_dir_2 = TaskGroup(group_id="sub_dir_2")
+    sub_dir_1_1 = TaskGroup(group_id="sub_dir_1_1")
+    with sub_dir_1, sub_dir_1_1:
+        sub_dir_1_sub_dir_1_1_b = aql.transform_file(
```

Comment:
@feluelle if our ultimate goal with transform_file over render is readability, I think we need to generate better names. I don't think a user would be able to parse what any of this means.

Reply:
I think that the name should be generated based on the file name, not the taskgroup.

Also why are these tasks under two task groups?

Reply:
> @feluelle if our ultimate goal with transform_file over render is readability, I think we need to generate better names. I don't think a user would be able to parse what any of this means.

This automatically generated code from the file names and the directory names. We can change it easily. I agree with you that we should not use this in official examples in our repository. Do you have some examples in mind? :)

> I think that the name should be generated based on the file name, not the taskgroup.

Then it won't be possible to use the same filename twice. That is what I want to show with this example. It contains two `a.sql` in different folders/task groups.

> Also why are these tasks under two task groups?

Because the file is in a **nested** directory, it will be in nested Task Groups. I think this is pretty useful when you have lots of tables to be able to create nested structures.